ctapipe/image/cleaning.py

```diff
@@ -49,40 +49,45 @@ def tailcuts_clean(geom, image, picture_thresh=7, boundary_thresh=5,
     `image[~mask] = 0`
 
     """
-    pixels_above_picture = image >= picture_thresh
+    # core_pixels: I call pixels, which are above the so called "picture_thresh"
+    # edge_pixels: I call pixels, which are above the so called "boundary_thresh"
+
+
+    core_pixels = image >= picture_thresh
 
     if keep_isolated_pixels or min_number_picture_neighbors == 0:
-        pixels_in_picture = pixels_above_picture
+        pixels_in_picture = core_pixels
     else:
         # Require at least min_number_picture_neighbors. Otherwise, the pixel
         #  is not selected
-        number_of_neighbors_above_picture = np.sum(pixels_above_picture &
-                                                   geom.neighbor_matrix, axis=1)
-        pixels_in_picture = pixels_above_picture & (
-            number_of_neighbors_above_picture >= min_number_picture_neighbors
+        number_of_core_neighbors = np.sum(
+            core_pixels & geom.neighbor_matrix,
+            axis=1
         )
+        enough_core_neighbors = number_of_core_neighbors >= min_number_picture_neighbors
+        pixels_in_picture = core_pixels & enough_core_neighbors
 
     # by broadcasting together pixels_in_picture (1d) with the neighbor
     # matrix (2d), we find all pixels that are above the boundary threshold
     # AND have any neighbor that is in the picture
-    pixels_above_boundary = image >= boundary_thresh
+    edge_pixels = image >= boundary_thresh
```

Comment:
calling this edge pixels may be confusing since we use that term to mean pixels on the edge of the camera elsewhere

Reply:
the term image "boundary" comes from the original implementation of this cleaning method, which was called "picture/boundary cleaning", so we should keep that name since people know what it means. So perhaps just call this `boundary_pixels`.  The name "tailcuts" came later, as a renaming of the same algorithm.